run_playlist_generator.py

```diff
@@ -0,0 +1,116 @@
+device = "cuda" if torch.cuda.is_available() else "cpu"
+
+caption_model = VisionEncoderDecoderModel.from_pretrained("nlpconnect/vit-gpt2-image-captioning").to(device)
+caption_tokenizer = GPT2TokenizerFast.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+caption_image_processor = ViTImageProcessor.from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+
+def get_caption(image_path):
+    
+    if os.path.exists(image_path):
+        image = Image.open(image_path)
+    else:
+        raise FileNotFoundError(f"Image not found at path: {image_path}")
+    
+    img = caption_image_processor(image, return_tensors="pt").to(device)
+    output = caption_model.generate(**img)
+    caption = caption_tokenizer.batch_decode(output, skip_special_tokens=True)[0]
+    return caption
+
+def load_state_dict(model, state_dict):
+    """
+    Load state_dict into the model.
+    """
+    model_state_dict = model.state_dict()
+    new_state_dict = {}
+    for key in state_dict.keys():
+        # Modify the key to match the model's keys
+        new_key = key.replace("image_layer.0", "image_layer").replace("fusion_linear.0", "fusion_linear")
+        if new_key in model_state_dict:
+            new_state_dict[new_key] = state_dict[key]
+    model.load_state_dict(new_state_dict)
+
+def determine_mood(image_path, caption):
+    model_path = "./2_image_caption_to_mood"
+    # Initialize tokenizer and configuration
+    config = BertConfig.from_pretrained('bert-base-uncased')
+    tokenizer_config = AutoConfig.from_pretrained("ayoubkirouane/BERT-Emotions-Classifier")
+
+    # Load tokenizer from files
+    tokenizer = AutoTokenizer.from_pretrained(
+    "bert-base-uncased",
+    config=tokenizer_config,
+    local_files_only=True  # Load tokenizer files from local directory
+    )
+
+    # Initialize model with correct arguments
+    resnet_model = resnet18(pretrained=True)
+    model = MultiModalEarlyFusionBertModel(config, num_classes=9, cnn_model=resnet_model)
+    
+    # Load model weights
+    model_state_dict = torch.load(os.path.join(model_path, "model_weights.pth"), map_location=torch.device('cpu'))
+    load_state_dict(model, model_state_dict)
+
+    outputs = tokenizer(caption, padding=True, truncation=True, return_tensors='pt')
+    input_ids = outputs ['input_ids']
+
+    image = Image.open(image_path).convert('RGB')
+    transform = transforms.Compose([
+        transforms.Resize((224,224)),
+        transforms.ToTensor(),
+    ])
+
+    image = transform(image).unsqueeze(0)
+
+    model = model.to(device)
+    image = image.to(device)
+    input_ids = input_ids.to(device)
+
+    # Forward pass, get logits or predictions
+    with torch.no_grad():
+        outputs = model.forward(input_ids=input_ids, images=image)
+
+    # get predicted mood
+    logits = outputs['logits']
+    predicted_class_id = logits.argmax(-1).item()
+
+    return predicted_class_id
+
+
+def main():
+    # step 1: generate caption based on images
```

Comment:
Could you make it so that it takes in like 4-9 images? That way we have more moods to work with for the playlist and does the actual mood board part?

Reply:
Yup! Will add the test to take in multiple images, especially since we need it for the analysis of the models